### ISSUE TITLE
NVDA ignores TH's aria-label and aria-labelledby

### ISSUE THREAD

**Issue by wkeese** [nvaccess/nvda#18225]
### Steps to reproduce:

1. Load the following in Chrome or another browser.

```html
<html>
<body>
<h1>Test screen reader with TH aria-labelledby</h1>
<p>
    When tabbing to the table's data cells, NVDA should announce the column names as "banana 1" and "banana 2" not "tomato 1" and "tomato 2".
    But it ignores the aria-label and aria-labelledby attributes, just using the text content
</p>
<table>
    <tr>
        <th aria-label="banana 1">tomato 1</th>
        <th aria-labelledby="b2">tomato 2</th>
    </tr>
    <tr>
        <td><button>cell 1</button></td>
        <td><button>cell 2</button></td>
    </tr>
</table>
<span id="b2">banana 2</span>
</body>
</html>
```

2. Tab to the buttons in the table.

### Actual behavior:

- Tomato 1 Column 1 Cell 1 button
- Tomato 2 Column 2 Cell 2 button

### Expected behavior:

- Banana 1 Column 1 Cell 1 button
- Banana 2 Column 2 Cell 2 button

#### NVDA version:

2024.4.2

#### Windows version:

Windows 11

#### Does the issue still occur after restarting your computer?

Yes.

#### If NVDA add-ons are disabled, is your problem still occurring?

I have no way to do that, there's no NVDA menu.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have no way to do that, there's no NVDA menu.
